src/OpenSpaceToolkit/Astrodynamics/Access/Generator.cpp

```diff
@@ -538,6 +538,60 @@ Generator                       Generator::AerRanges                        (
 
 }
 
+Generator                       Generator::AerMask                           (  const   Map<Real, Real>&            anAzimuthElevationMask,
+                                                                                const   Interval<Real>&             aRangeRange,
+                                                                                const   Environment&                anEnvironment                               )
+{                    
+
+    using ostk::core::types::Real ;
+    using ostk::core::ctnr::Map ;
+
+    if ((anAzimuthElevationMask.empty()) || (anAzimuthElevationMask.begin()->first < 0.0) || (anAzimuthElevationMask.rbegin()->first > 360.0))
+    {
+        throw ostk::core::error::runtime::Wrong("Azimuth-Elevation Mask") ;
+    }
+
+    for (const auto& azimuthElevationPair : anAzimuthElevationMask) {
+        if ((azimuthElevationPair.second).abs() > 90.0) 
+        {
+            throw ostk::core::error::runtime::Wrong("Azimuth-Elevation Mask") ;
+        }
+    }
+
+    Map<Real, Real> anAzimuthElevationMask_deg = anAzimuthElevationMask ;
+    const Interval<Real> rangeRange_m = aRangeRange ;
+
+    if (anAzimuthElevationMask_deg.begin()->first != 0.0)
+    {
+        anAzimuthElevationMask_deg.insert({ 0.0, anAzimuthElevationMask_deg.begin()->second }) ;
+    }
+
+    if (anAzimuthElevationMask_deg.rbegin()->first != 360.0)
+    {
+        anAzimuthElevationMask_deg.insert({ 360.0, anAzimuthElevationMask_deg.begin()->second }) ;
+    }
+
+    const std::function<bool (const AER&)> aerFilter = [anAzimuthElevationMask_deg, rangeRange_m] (const AER& anAER) -> bool
+    {
+
+        const Real azimuth = anAER.getAzimuth().inDegrees(0.0, +360.0) ;
+        const Real elevation = anAER.getElevation().inDegrees(-180.0, +180.0) ;
+
+        auto itLow = anAzimuthElevationMask_deg.lower_bound(azimuth) ; itLow-- ;
+        auto itUp = anAzimuthElevationMask_deg.upper_bound(azimuth) ;
+
+        const std::vector<Real> lowToUpVector = { itUp->first - itLow->first, itUp->second - itLow->second } ;
+        const std::vector<Real> lowToPointVector = { azimuth - itLow->first, elevation - itLow->second } ;
+
+        return (lowToUpVector[0] * lowToPointVector[1] - lowToUpVector[1] * lowToPointVector[0] > 0.0)
```

Comment:
Could you explain the calculation?

Reply:
(adding a comment for future reference would be nice)

Reply:
Comments added. 

The mask defines a function f : azimuth -> elevation. This determines whether the tested {azimuth, elevation} point lies above or below this function.

`lowToUpVector` represents the vector between the two mask data points with azimuth values bounding the tested azimuth. `lowToPointVector` goes from the data point with azimuth lower bound to the tested point. If the determinant of these two vectors is positive, the tested point lies above the function defined by the mask - we select it.